examples/smqtk/smqtk_list.yaml

```diff
@@ -0,0 +1,13 @@
+foo:
+    - smqtk: 
+        class: HelloWorldSmqtk
+        config:
+            foo: 21
+            bar: 22
+            baz: 23
+    - smqtk: 
+        class: HelloWorldSmqtk
```

Comment:
The `smqtk_core.configuration` module [documents a similar schema](https://github.com/Kitware/SMQTK-Core/blob/master/smqtk_core/configuration.py#L9) to the one you're defining here.

I 100% would love to have a more formal and better schema defined in SMQTK for usage within that realm (which I argue here a chunk of what going on here probably should).

Reply:
Are you making an actionable suggestion here? We also need to lock down the schema for the `smqtk` directive, and a lot of what you wrote above can influence the final shape of it.

@Pikus16, this is another point of evidence that we should perhaps stop and write up a more formal design RFC on how we understand this directive to work.

Reply:
I suppose an actionable suggestion is that the `smqtk` directive here follows the schema loosely defined within smqtk-core, with schema modifications and improvements happening in smqtk-core. My motivation for this suggesting would be to reduce (prevent?) the creation of additional forms/syntaxes of configuring SMQTK types outside of smqtk-core itself. This kind of action is pretty central to the intention of smqtk-core and the synergy of Pluggable and Configurable. If the current solution there is not appropriate for your use-case I would be interested in fixing/updating it towards being so.

Reply:
Alright, well that pretty much seals it: we really should write up an RFC for how this should work going forward.

I'd rather not hold up this PR to solve it right away, since we have nothing right now, and a working prototype of the feature would be a first step from which to launch iterative refinement. So, let's land this PR and then file an issue directing us to think about achieving better synergy with SMQTK's own offerings.

Reply:
Sounds good to me. I'm happy enough to get it on the table.

Reply:
See #141.